### PR TITLE
[sil-combine] Use canonicalizeOSSALifetimes to eliminate unnecessary copies inserted due to RAUWing

### PR DESCRIPTION
**Pull request by gottesmm**
This is the first in a series of patches that use canonicalizeOSSALifetimes to eliminate extra copies inserted. canonicalizeOSSALifetimes is a new helper that takes in a list of defs found via CanonicalizeOSSALifetime::getCanonicalCopiedDef(...) and canonicalizes them. I added it in SILCombine to canonicalize away any copies inserted due to us RAUWing values using the OwnershipRAUWHelper.

I am going to add it into other passes where we are creating new ownership calls (consider mem2reg). This will eliminate phase ordering issues in the compiler.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
@swift-ci benchmark

**Comment by gottesmm** [comment]
@swift-ci benchmark

**Comment by gottesmm** [comment]
I am not expecting any benchmark changes. What I am trying to just do is teach individual passes that use the Ownership RAUW infrastructure how to use copy propagation to eliminate unnecessary copies by canonicalizing lifetimes.

**Comment by gottesmm** [comment]
Regressions are different. Its most likely noise.

File: lib/SILOptimizer/SILCombiner/SILCombine.cpp [hunk 1]
```diff
@@ -299,10 +302,39 @@ bool SILCombiner::doOneIteration(SILFunction &F, unsigned Iteration) {
...
-    // its contents to the worklist and then clear said list in preparation for
-    // the next iteration.
+    // SILBuilder during this iteration. In order to finish this round of
+    // SILCombine, go through the tracking list and add its contents to the
+    // worklist and then clear said list in preparation for the next
+    // iteration. We canonicalize any copies that we created in order to
+    // eliminate unnecessary copies introduced by RAUWing when ownership is
+    // enabled.
+    //
+    // NOTE: It is ok if copy propagation results in MadeChanges being set to
```
Comment (atrick) [inline lib/SILOptimizer/SILCombiner/SILCombine.cpp:312]: Someone reading this comment won't realize that CanonicalizeOSSALifetime is mainly doing copy propagation.

**Review by atrick — APPROVED** [review]
lgtm

**Pull request merged**